aws/ecs_cluster.go

```diff
@@ -11,14 +11,63 @@ import (
 	"github.com/gruntwork-io/gruntwork-cli/errors"
 )
 
+// Used in this context to determine if the ECS Cluster is ready to be used & tagged
+// For more details on other valid status values: https://docs.aws.amazon.com/sdk-for-go/api/service/ecs/#Cluster
+const activeEcsClusterStatus string = "ACTIVE"
+const provisioningEcsClusterStatus = "PROVISIONING"
+
+// filter out all active ecs clusters
+func getAllActiveEcsClusterArns(awsSession *session.Session) ([]*string, error) {
+	svc := ecs.New(awsSession)
+
+	allClusters, err := getAllEcsClusters(awsSession)
+	if err != nil {
+		logging.Logger.Errorf("Error getting all ECS clusters")
+		return nil, errors.WithStackTrace(err)
+	}
+
+	var result []*ecs.Cluster
+
+	for _, cluster := range allClusters {
+		input := &ecs.DescribeClustersInput{
+			Clusters: []*string{cluster},
```

Comment:
Instead of making one API call per ECS cluster, it looks like `Clusters` accepts a slice, so could you describe all of them? Note: there may be some limit on how many `Clusters` you can query at a time, so you may have to batch these a bit, but it would still be faster than one API call at a time.

Reply:
Lol race condition! Had the same comment, and you can see https://github.com/gruntwork-io/cloud-nuke/pull/146#discussion_r505672148 for a suggested fix.